cmd/kured/main.go

```diff
@@ -333,19 +343,27 @@ func drain(client *kubernetes.Clientset, node *v1.Node) {
 
 	drainer := &kubectldrain.Helper{
 		Client:              client,
-		GracePeriodSeconds:  -1,
+		Ctx:                 context.Background(),
+		GracePeriodSeconds:  drainGracePeriod,
 		Force:               true,
 		DeleteEmptyDirData:  true,
 		IgnoreAllDaemonSets: true,
 		ErrOut:              os.Stderr,
 		Out:                 os.Stdout,
 	}
+	if forceReboot {
+		drainer.Timeout = forceTimeout
+	}
+
 	if err := kubectldrain.RunCordonOrUncordon(drainer, node, true); err != nil {
```

Comment:
Wouldn't `--force-reboot` imply that we want to reboot even if cordon fails?

Reply:
The control flow here is a bit hard to follow here, but my understanding is that setting the `drainer.Timeout` field causes the `kubectldrain.RunNodeDrain` to err after the timeout has expired. It's a bit hard to spot, but the `log.Fatalf` does the actual work of panicing normally, and so setting `forceReboot` bypasses this panic, and so `drain` returns and `invokeReboot` is called.

Reply:
I'm talking about what to do in the err condition for invoking `kubectldrain.RunCordonOrUncordon`. I think `--force-reboot` should indicate that we want to reboot even if the node was not able to be marked as unschedulable, similar to how we are rebooting even if we aren't able to drain the node.

Basically, I think `--force-reboot` makes the most sense if *both* cordon and drain are best-effort, but not strictly required.

Reply:
Oh I see. That makes sense to me. I will update the condition here to also not panic if force-reboot is set.